example/main-line-layer.js

```diff
@@ -0,0 +1,254 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/* eslint-disable func-style */
+/* eslint-disable no-console */
+/* global console, process */
+import 'babel-polyfill';
+import document from 'global/document';
+import window from 'global/window';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {createStore} from 'redux';
+import {Provider, connect} from 'react-redux';
+import autobind from 'autobind-decorator';
+
+import MapboxGLMap from 'react-map-gl';
+import OverlayControl from './overlay-control';
+
+import request from 'd3-request';
+import {
+  DeckGLOverlay,
+  LineLayer
+} from '../src';
+
+// ---- Default Settings ---- //
+/* eslint-disable no-process-env */
+const MAPBOX_ACCESS_TOKEN = process.env.MAPBOX_ACCESS_TOKEN ||
+  'Set MAPBOX_ACCESS_TOKEN environment variable or put your token here.';
+
+const GEO_JSON_FILE = './example/data/your.geo.json';
+const INITIAL_STATE = {
+  mapViewState: {
+    latitude: 21.1250,
+    longitude: -101.6860,
+    zoom: 11.5
+  },
+  lines: null
+};
+
+// ---- Action ---- //
+function updateMap(mapViewState) {
+  return {type: 'UPDATE_MAP', mapViewState};
+}
+
+function loadGeoJson(data) {
+  return {type: 'LOAD_GEO_JSON', geoJson: data};
+}
+
+function loadPoints(points) {
+  return {type: 'LOAD_POINTS', points};
+}
+
+// ---- Reducer ---- //
+function reducer(state = INITIAL_STATE, action) {
+  switch (action.type) {
+  case 'UPDATE_MAP':
+    return {...state, mapViewState: action.mapViewState};
+  case 'LOAD_GEO_JSON':
+    const colors = [[255, 0, 0], [0, 128, 0], [0, 0, 255]];
+    let i = 0;
+    const lines = [];
+    action.geoJson.features.forEach(f => {
+      i++;
+      const coordinates = f.geometry.coordinates;
+      let prev = coordinates[0];
+      coordinates.slice(1).forEach(c => {
+        lines.push({
+          position: {
+            x0: prev[0],
+            y0: prev[1],
+            x1: c[0],
+            y1: c[1]
+          },
+          color: colors[i % 3]
+        });
+        prev = c;
+      });
+    });
+    return {...state, lines};
```

Comment:
line 77-96 show how to derive the lines data from a geoJson file, it is valuable if we have the geoJson file included in the example. for this PR, shall we only merge the line-layer part and modify the current example/main.js, and remove the separate main-line-layer.js file?